src/sagemaker_xgboost_container/metrics/custom_metrics.py

```diff
@@ -50,6 +51,21 @@ def accuracy(preds, dtrain):
     return 'accuracy', score
 
 
+def balanced_accuracy(preds, dtrain):
```

Comment:
You can define a base function `compute_score(preds, dtrain) -> score` and then simplify all of these functions to just 

```python
def balanced_accuracy(preds, dtrain):
  return 'balanced_accuracy', compute_score(preds, dtrain)
```